### PR TITLE
Port bridge

### PR DESCRIPTION
**Pull request by matt001k**
Ports bridge app to use bm_core
Adds extern directive to bm_core header files
Adds documentation for supported services in middleware

### CODE REVIEW

File: src/bm_service_request.c [hunk 1]
```diff
@@ -85,7 +85,7 @@ bool bm_service_request(size_t service_strlen, const char *service,
   if (!service) {
     return rval;
   }
-  if (!data_len || !data) {
```
Comment (towynlin) [inline src/bm_service_request.c:88]: This is a change in the logic and looks incorrect. We want to return false (failure) early if the `data_len` is zero, right?

Comment (matt001k) [reply src/bm_service_request.c:88]: No if you compare it to the logic in `bm_protocol` we can actually pass in a NULL pointer here if data length is zero. This was the bug I was mentioning this morning that was introduced.

Comment (matt001k) [reply src/bm_service_request.c:88]: snippet from `sys_info_service`:
```
bool sys_info_service_request(uint64_t target_node_id,
                              BmServiceReplyCb reply_cb, uint32_t timeout_s) {
  bool rval = false;
  char *target_service_str = (char *)bm_malloc(BM_SERVICE_MAX_SERVICE_STRLEN);
  if (target_service_str) {
    size_t target_service_strlen =
        snprintf(target_service_str, BM_SERVICE_MAX_SERVICE_STRLEN,
                 "%016" PRIx64 "%s", target_node_id, sys_info_service_suffix);
    if (target_service_strlen > 0) {
      rval = bm_service_request(target_service_strlen, target_service_str, 0,
                                NULL, reply_cb, timeout_s);
    }
  }
  bm_free(target_service_str);
  return rval;
}
```

Comment (towynlin) [reply src/bm_service_request.c:88]: Ah, I see, nice. So this is for sending a request with no payload. 👍

**Review by towynlin — APPROVED** [review]

**Pull request merged**